packages/frontend/editor-ui/src/features/workflows/canvas/composables/useCanvasMapping.ts

```diff
@@ -711,7 +763,14 @@ export function useCanvasMapping({
 		} else if (nodeHasIssuesById.value[connection.source]) {
 			status = 'error';
 		} else if (runDataTotal > 0 && lastSourceTask?.executionStatus !== 'canceled') {
-			status = 'success';
+			// For non-main connections (model, memory, tool, etc.), only mark as executed
+			// if the target node also executed, since these are passive connections
+			const isMainConnection = type === NodeConnectionTypes.Main;
+			const targetNodeHasAnyExecution = nodeExecutionRunDataById.value[connection.target];
+
+			if (isMainConnection || targetNodeHasAnyExecution) {
```

Comment:
Wouldn't we need to check that it's not a main connection, so that we don't incorrectly flag a main exec as successful based on the connecting node?

```suggestion
			if (isMainConnection || !isMainConnection && targetNodeHasAnyExecution) {
```